src/plugins/discover/public/application/angular/helpers/row_formatter.ts

```diff
@@ -48,9 +50,11 @@ export const formatTopLevelObject = (
   const sorted = Object.entries(fields).sort(([keyA], [keyB]) => keyA.localeCompare(keyB));
   sorted.forEach(([key, values]) => {
     const field = indexPattern.getFieldByName(key);
+    const displayKey = fields.getByName ? fields.getByName(key)?.displayName : undefined;
```

Comment:
QQ: why is `displayKey` now needed? thx!

Reply:
In a Document column, if there is a custom label for the field, we would want to use that one, instead of the actual field name. `displayName` on the index pattern field returns this value, while `key` is just the field name.

Reply:
makes sense! thx!